lua/codeium/io.lua

```diff
@@ -304,37 +299,48 @@ function M.job(cmd)
 end
 
 function M.generate_uuid()
-	-- TODO: windows
+	local uuid
+	if vim.fn.has("win32") then
+		uuid = string.gsub("xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx", "[xy]", function(c)
+			return string.format("%x", (c == "x") and (math.random(16) - 1) or (((math.random(16) - 1) % 4) + 8))
+		end)
+	end
 	if not M.executable("uuidgen") then
-		if M.get_system_info().os == "windows" then
-			error("TODO(uuidgen is currently required, even on Windows)")
-		else
-			error("uuiden could not be found")
-		end
+		error("uuiden could not be found")
 	else
-		local uuid, err = M.get_command_output("uuidgen")
+		local err
+		uuid, err = M.get_command_output("uuidgen")
 		if err then
 			error("failed to generate UUID: " .. vim.inspect(err))
 		end
-		return uuid
 	end
+	return uuid
 end
 
 function M.gunzip(path, callback)
-	-- TODO: windows
-	if not M.executable("gzip") then
-		if M.get_system_info().os == "windows" then
-			callback(nil, "TODO(gzip is currently required, even on Windows)")
-		else
-			callback(nil, "gzip could not be found")
-		end
-	else
+	if vim.fn.has("win32") then
 		M.job({
-			"gzip",
-			"-d",
+			"powershell.exe",
+			"-noprofile",
+			"-command",
+			"\"Expand-Archive -Path '",
```

Comment:
I would expect that it would require something along the lines of:

```lua
// ...
"-command",
"Expand-Archive -Path '" .. path .. "' -DestinationPath '.'",
// ...
```

`job` doesn't concatenate args, parsing native APIs take args as an array, even on Windows, and the shell is responsible for parsing them out to an array. That being said nvim could be doing some nasty hack with `cmd.exe` internally.

Reply:
I have no idea if nvim is doing anything strange. I decided to go with powershell, though it could have been cmd just as easily. It seems to not blow up, so I went with it, it doesn't come from an informed opinion. I'm pretty unfamiliar with shells across the board, have only written a couple of powershell scripts and and BAT files over the years, so as far as the details regarding how it does or does not take args and what it does with them I'm just taking guesses and asking some AI to throw me a bone. I think I just saw that the job call with gzip took the args like this, so I just tried to make it similar. If you feel like it's better another way I'll gladly do it, I defer to anyone else who knows what their doing, lol.